src/movies.js

```diff
@@ -1,25 +1,68 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors() {
+  let moviedirectors = movies.map(function(e){
+    return e.director
+  })
+return moviedirectors
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies() {
+  let dramaMoviesBySteveSpielberg = movies.filter(function(el){
+    return el.director == 'Steven Spielberg' && el.genre.includes('Drama') == true
+  })
+  return dramaMoviesBySteveSpielberg.length
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+  let averageScore = movies.map(function(el){
+    return el.score
+
+  })
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function scoresAverage() {
+  let averageScore = movies.map(function(el){
+    return el.score})
+  
+  let countofscore = averageScore.length
+  
+  let summedscore = averageScore.reduce(function(a,b){
+    return a+b
+  })
+  return (summedscore/countofscore).toFixed(2)
+  
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear() {
+  sortedArr = movies.sort(function(a,b){
```

Comment:
Here you should be using a let or const. This way of declaring is possible, but should not really be used as it can cause problems.

https://stackoverflow.com/questions/49357522/is-there-a-use-for-not-using-var-let-or-const